### ISSUE TITLE
Load jsforce within OpenSource LWC app fails

### ISSUE THREAD

**Issue by novumcoder** [jsforce/jsforce#1179]
Hi, i am creating a simple LWC based app for a website using the lwc.dev opensource one from Salesforce to build web components.
Using jsforce 1.10.0 but also 2.0.0-beta.8 fails when the demo LWC app is trying to load the jsforce library.
I attached the screenshot below showing the error thrown. Any idea which this is happening?

![image](https://user-images.githubusercontent.com/6917790/133069862-a674d41f-696d-4403-924f-1f28f5eb2727.png)

Second screenshot shows the only change i put into the dummy LWC in order to load the library, which seems to be failing.

![image](https://user-images.githubusercontent.com/6917790/133069986-2c3cdb55-fb40-471b-ad34-2a66aec10eed.png)

**Comment by paustint** [comment]
Do you have the same results if you try `import * as jsforce from 'jsforce'`?

I will try to play around with it to see if I can get it going as well.

**Comment by novumcoder** [comment]
Yes i tried all variants. All gave me the same error. I tried the following so far, all ending with the same error:

`import * as jsforce from 'jsforce'`
`import jsforce from 'jsforce'`
`import { jsforce } from 'jsforce'`

**Comment by paustint** [comment]
@novumcoder - In general, I would HIGHLY recommend avoiding using jsforce directly in your browser and instead manage it on the server (this is what I do for https://getjetstream.app/ )

1. jsforce makes cross-domain requests, which are blocked by default by the browser due-to CORS
2. In the browser, you are exposing all sensitive login information to the webpage, which should be avoided in most situations

With that said, I got a project working, but it was extremely painful... I forgot how non-existent / unhelpful the LWC open source docs are regarding project configuration. 🥇 

You can check out the project here: https://github.com/paustint/lwc-jsforce-test
(NOTE: jsforce fails because of CORS, but you can see that it is making requests and getting back errors. I even hard-coded a password to a dev or in the codebase to make it easy ;) )

**I don't know why you had issues - but I did not have any issues importing or invoking the library.**

jsforce uses node APIs in the codebase in a way which requires browser polyfills if using webpack, and webpack 5 no longer ships with polyfills built-in.

I had to figure out how to include the proper polyfills to be compatible with webpack 5, and that proved to be a bit challenging - this requires a custom webpack config, a few extra packages installed, and flags in package.json to use the custom webpack config.

Aside from that,

**Comment by novumcoder** [comment]
Hey, that makes absolutely sense. I gonna move the jsforce part to my API running on AWS already, i solved the CORS issue by an jsforce-proxy API running on AWS. Your example project is working and i see why, it is because of the browserify plugins.

Thanks for the hint.

**Comment by nirajshelke** [comment]
Hello,

I am facing an issue **Uncaught ReferenceError: process is not defined at Object../node_modules/readable-stream/lib/_stream_writable.js**  when I am importing jsforce in component. I have attached the Screenshot below

![Screenshot from 2022-02-28 22-38-46](https://user-images.githubusercontent.com/73650079/156027293-df52439e-f18c-4d6f-8d38-d14a3a4c4777.png)

Can someone help me to resolve this issue?
